xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/WcagContext.java

```diff
@@ -0,0 +1,413 @@
+            entry("listitem", false),
+            entry("marquee", false),
+            entry("meta-refresh", false),
+            entry("meta-viewport", false),
+            entry("nested-interactive", false),
+            entry("no-autoplay-audio", false),
+            entry("object-alt", false),
+            entry("role-img-alt", false),
+            entry("scrollable-region-focusable", false),
+            entry("select-name", false),
+            entry("server-side-image-map", false),
+            entry("svg-img-alt", false),
+            entry("td-headers-attr", false),
+            entry("th-has-data-cells", false),
+            entry("valid-lang", false),
+            entry("video-caption", false),
+
+            // WCAG 2.1 Level A & AA Rules
+            entry("autocomplete-valid", false),
+            entry("avoid-inline-spacing", false),
+
+            // WCAG 2.2 Level A & AA Rules
+            entry("target-size", false)
+    );
+
+    /**
+     * Tags to take into account during axe-core validation.
+     * All rules with at least one of those tags will be validated.
+     */
+    private static final List<String> VALIDATE_TAGS = Arrays.asList("wcag2a", "wcag2aa", "wcag21a", "wcag21aa");
+
+    /**
+     * Rules to disable during axe-core validation.
+     */
+    private static final List<String> DISABLED_RULES = List.of();
+    private static String shortFailReport = "";
+    /**
+     * Stores the result of an axe-core validity scan.
+     */
+    private static final class WCAGTestResults
+    {
+        private String failReport = "";
+        private String warnReport = "";
+        /**
+         * @param testMethodName in which the validation happened.
+         * @param url from which these results have been generated
+         * @param pageClassName class of the page validated
+         * @param axeResults results of the validation
+         */
+        private WCAGTestResults(String testMethodName, String url, String pageClassName, Results axeResults)
+        {
+            // Generate the report as soon as the results are in.
+            List<Rule> failingViolations = axeResults.getViolations()
+                .stream()
+                .filter(rule -> !FAILS_ON_RULE.containsKey(rule.getId()) || FAILS_ON_RULE.get(rule.getId()))
+                /* If the ruleid is not defined in FAILS_ON_RULE,
+                the default behavior will be to add it to the fails.
+                In order to resolve these test-suite fails quickly, set them as "false" in FAILS_ON_RULE.
+                 */
+                .collect(Collectors.toList());
+            if (XWikiCustomAxeReporter.getReadableAxeResults(testMethodName, pageClassName, url, failingViolations))
+            {
+                this.failReport = XWikiCustomAxeReporter.getAxeResultString();
+                if (shortFailReport.isEmpty())
+                {
+                    /* Only store the first failed rule to display in the error logs. */
```

Comment:
I know this was my suggestion, we could imho also just display all failed rules if this requires extra work to get the first result. My intention was primarily to display something informative when a test fails such that the developer doesn't first need to get the log to see the reason for the failure. My understanding is that we will only enable rules for which we don't have known failures so in this case this also shouldn't flood the logs unless a regression is introduced (or a new test is added, but then it should be fixed immediately).

Reply:
The thing is, for less memory use, I process reports as soon as they are generated so I only keep the string we want. This is less flexible than keeping everything in memory and waiting for the end to generate reports.
This makes it a bit weird to save only the first rule break, since it's often only a part of one validation check.
It'd be easier to display the first report fully, however it could get quite lengthy if a few rules fail at once.
This shouldn't happen too often, so I'll remove this static variable and go with the easy solution

Reply:
Sounds good, we can change it also later if we should discover that it is indeed flooding the log.